### PR TITLE
Fix errors in class definitions and add error handling to create_report methods

### PR DESCRIPTION
**Pull request by noenfugler**
1) Some minor fixes to type definitions in CommentReportView and PostReport classes.
2) Added additional error handling to create_report in enable more graceful error handling in calling code.

### CODE REVIEW

File: pylemmy/models/comment.py [hunk 1]
```diff
@@ -93,8 +93,10 @@ def create_report(self, reason: str) -> CommentReport:
...
 
-        return CommentReport(
-            lemmy=self.lemmy, report=parsed_result.comment_report_view, comment=self
-        )
+        if not 'error' in result:
+            parsed_result = api.comment.CommentReportResponse(**result)
+            return CommentReport(
+                lemmy=self.lemmy, report=parsed_result.comment_report_view, comment=self)
+        else:
+            raise result['error']
```
Comment (dcferreira) [inline pylemmy/models/comment.py]: I don't think this is where the exception should be raised, but rather in `lemmy.post_request`.

Here's a suggestion (to change in `pylemmy/lemmy.py`:
```python
    def post_request(
        self,
        path: LemmyAPI,
        params: Optional[BaseApiModel] = None,
    ):
        """Send a POST request to the desired path.

        :param path: A Lemmy endpoint.
        :param params: Parameters to send with the request (in the body).
        """
        response = self.session.post(
            self._get_url(path),
            json=params.dict() if params is not None else {},
            timeout=self.request_timeout,
        )
        response.raise_for_status()
        return response.json()
```

Actually the same should also be done for `get_request`. Could you please implement that @noenfugler ?

File: pylemmy/models/post.py [hunk 2]
```diff
@@ -119,8 +119,11 @@ def create_report(self, reason: str) -> PostReport:
...
 
-        return PostReport(
-            lemmy=self.lemmy, report=parsed_result.post_report_view, post=self
-        )
+        if not 'error' in result:
+            parsed_result = api.post.PostReportResponse(**result)
+            return PostReport(
+                lemmy=self.lemmy, report=parsed_result.post_report_view, post=self)
+        else:
+            raise result['error']
```
Comment (dcferreira) [inline pylemmy/models/post.py]: Same as the other comment, don't think the check for error should be here.

**Review by dcferreira — CHANGES_REQUESTED** [review]
Thanks again for the contribution @noenfugler . I have some suggestions to consider, could you have a look in the comments below?

File: pylemmy/models/comment.py (hunk 1 above)
Comment (noenfugler) [reply pylemmy/models/comment.py]: Happy to.  Hopefully I don't mess up the Github workflow.  I'm on a bit of a learning curve here.

**Comment by noenfugler** [comment]
Changes made.  Ready for re-review.

**Review by dcferreira — APPROVED** [review]
Perfect, thanks a lot! :)

**Pull request merged**